api/src/main/java/org/pmiops/workbench/dataset/GenomicDatasetServiceImpl.java

```diff
@@ -0,0 +1,142 @@
+package org.pmiops.workbench.dataset;
```

Comment:
This is all moved or copied from DataSetService, plus a few modernizations like ImmutableList -> List.  No intended changes in functionality.